src/apps/api/tests/test_datasets.py

```diff
@@ -0,0 +1,36 @@
+import factory
+
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from datasets.models import Data
+from factories import UserFactory, DataFactory
+
+
+class DatasetAPITests(APITestCase):
+    def setUp(self):
+        self.creator = UserFactory(username='creator', password='creator')
+
+        self.existing_dataset = DataFactory(created_by=self.creator, name="Test!")
+
+    def test_dataset_api_checks_duplicate_names_for_same_user(self):
+        self.client.login(username='creator', password='creator')
+
+        # Make a dataset that conflicts with existing dataset name
+        resp = self.client.post(reverse("data-list"), {
+            'name': 'Test!',
+            'type': Data.COMPETITION_BUNDLE,
+            'request_sassy_file_name': factory.Faker('file_name'),
```

Comment:
does this work here? since this isn't inside of a factory? or does this do the same thing the username call was doing?

Reply:
good catch, ty, fixed ✔️